### PR TITLE
Fixed 1D regular sampling + tests and updates to FEniCS example

### PR DESCRIPTION
**Pull request by eecsu**
The 1D regular sampling was not producing arrays correctly. Now fixed. It works in 1D examples now (which we still do not have, so should probably add at some point). 

The FEniCS example had to be updated in just a few places to work with the newer version of FEniCS.

### CODE REVIEW

File: bet/postProcess/plotP.py [hunk 1]
```diff
@@ -214,14 +214,19 @@ def plot_1D_marginal_probs(marginals, bins, sample_set,
...
+                        0.02*1.05*np.max(marginals[i]/(bins[i][1]-bins[i][0])),
+                        'ko', markersize=15)
             if lambda_label is None:
                 label1 = r'$\lambda_{' + str(i+1) + '}$'
             else:
                 label1 = lambda_label[i]
-            ax.set_xlabel(label1) 
-            ax.set_ylabel(r'$\rho$')
-            plt.tight_layout()
+            #ax.set_xlabel(label1)
```
Comment (smattis) [inline bet/postProcess/plotP.py]: Is there a reason stuff is left commented out?

File: bet/postProcess/plotP.py [hunk 2]
```diff
@@ -440,3 +445,115 @@ def smooth_marginals_2D(marginals, bins, sigma=10.0):
...
+                plt.axis([plot_domain[i][0], plot_domain[i][1], 
+                          plot_domain[j][0], plot_domain[j][1]]) 
+            fig.savefig(filename + "_2D_contours_" + str(i) + "_" + str(j) +\
+                    file_extension, transparent=True)
+            if interactive:
+                plt.show()
+            else:
+                plt.close()
+ 
+    comm.barrier()
```
Comment (smattis) [inline bet/postProcess/plotP.py]: put newline at end of file

File: examples/FEniCS/BET_script.py [hunk 3]
```diff
@@ -97,12 +97,13 @@
 Q_ref = my_model(param_ref)
 
 # Create some plots of input and output discretizations
-plotD.scatter_2D(input_samples, ref_sample=param_ref[0,:],
-                 filename='FEniCS_ParameterSamples',
-                 file_extension = '.eps')
-if Q_ref.size == 2:
-    plotD.show_data_domain_2D(my_discretization, Q_ref=Q_ref[0,:],
-            file_extension=".eps")
+#plotD.scatter_2D(input_samples, ref_sample=param_ref[0,:],
```
Comment (smattis) [inline examples/FEniCS/BET_script.py]: Why is it left commented out?

File: examples/FEniCS/BET_script.py [hunk 4]
```diff
@@ -134,22 +135,22 @@
 marginals2D = plotP.smooth_marginals_2D(marginals2D, bins, sigma=0.5)
 
 # plot 2d marginals probs
-plotP.plot_2D_marginal_probs(marginals2D, bins, input_samples,
-                             filename="FEniCS",
-                             lam_ref=param_ref[0,:],
-                             file_extension=".eps",
-                             plot_surface=False)
+#plotP.plot_2D_marginal_probs(marginals2D, bins, input_samples,
```
Comment (smattis) [inline examples/FEniCS/BET_script.py]: Ditto

File: examples/FEniCS/myModel.py [hunk 5]
```diff
@@ -95,8 +96,8 @@ def eval(self, v, x):
         return v
 
     # Define the QoI maps
-    Chi_1 = CharFunc([0.75, 1.25, 7.75, 8.25])
-    Chi_2 = CharFunc([7.75, 8.25, 0.75, 1.25])
+    Chi_1 = CharFunc(degree=1, region = [0.75, 1.25, 7.75, 8.25])
```
Comment (smattis) [inline examples/FEniCS/myModel.py]: Don't put spaces around = in arguments.

File: test/test_postProcess/test_plotP.py [hunk 6]
```diff
@@ -202,3 +220,20 @@ def test_plot_marginals_2D(self):
...
+        marginals[(0,1)][0][1]*=2.0
+        try:
+            plotP.plot_2D_marginal_probs(marginals, bins, self.samples,
+                                         filename = "file", interactive=False)
+            go = True
+            if os.path.exists("file_2D_contours_0_1.png") and comm.rank == 0:
+                os.remove("file_2D_contours_0_1.png")
+        except (RuntimeError, TypeError, NameError):
+            go = False
+        nptest.assert_equal(go, True)
```
Comment (smattis) [inline test/test_postProcess/test_plotP.py]: put newline at end of file

File: bet/postProcess/plotP.py (hunk 1 above)
Comment (eecsu) [inline bet/postProcess/plotP.py]: I forgot to delete it. Deleting now...

File: bet/postProcess/plotP.py (hunk 2 above)
Comment (eecsu) [inline bet/postProcess/plotP.py]: yessir

File: test/test_postProcess/test_plotP.py (hunk 6 above)
Comment (eecsu) [inline test/test_postProcess/test_plotP.py]: yessir

File: examples/FEniCS/BET_script.py (hunk 3 above)
Comment (eecsu) [inline examples/FEniCS/BET_script.py]: I was testing it with inputs of more than 2D. Now I just check to see if the dimensions of the input space is 2D.

File: examples/FEniCS/BET_script.py (hunk 4 above)
Comment (eecsu) [inline examples/FEniCS/BET_script.py]: It was making too many plots when the dimension of the input space was high. Changed it back now.

File: examples/FEniCS/myModel.py (hunk 5 above)
Comment (eecsu) [inline examples/FEniCS/myModel.py]: Done

**Comment by smattis** [comment]
Ok. Looks good to me. I'm merging.

**Pull request merged**